monai/transforms/utils.py

```diff
@@ -208,3 +209,136 @@ def generate_pos_neg_label_crop_centers(label, size, num_samples, pos_ratio, ran
+    """
+    control grid with two additional point in each direction
+    """
+    grid_shape = []
+    for d, s in zip(spatial_shape, spacing):
+        d = int(d)
+        if d % 2 == 0:
+            grid_shape.append(np.ceil((d - 1.) / (2. * s) + 0.5) * 2. + 2.)
+        else:
+            grid_shape.append(np.ceil((d - 1.) / (2. * s)) * 2. + 3.)
+    return create_grid(grid_shape, spacing, homogeneous, dtype)
+
+
+def create_rotate(spatial_dims, radians):
+    """
+    create a 2D or 3D rotation matrix
+    Args:
+        spatial_dims (2|3): spatial rank
+        radians (float or a sequence of floats): rotation radians
+            when spatial_dims == 3, the `radians` sequence corresponds to
+            rotation in the 1st, 2nd, and 3rd dim respectively.
+    """
+    radians = ensure_tuple(radians)
+    if spatial_dims == 2:
+        if len(radians) >= 1:
+            sin_, cos_ = np.sin(radians[0]), np.cos(radians[0])
+            return np.array([[cos_, -sin_, 0.], [sin_, cos_, 0.], [0., 0., 1.]])
+
+    if spatial_dims == 3:
+        affine = None
+        if len(radians) >= 1:
+            sin_, cos_ = np.sin(radians[0]), np.cos(radians[0])
+            affine = np.array([
+                [1., 0., 0., 0.],
+                [0., cos_, -sin_, 0.],
+                [0., sin_, cos_, 0.],
+                [0., 0., 0., 1.],
+            ])
+        if len(radians) >= 2:
+            sin_, cos_ = np.sin(radians[1]), np.cos(radians[1])
+            affine = affine @ np.array([
+                [cos_, 0.0, sin_, 0.],
+                [0., 1., 0., 0.],
+                [-sin_, 0., cos_, 0.],
+                [0., 0., 0., 1.],
+            ])
+        if len(radians) >= 3:
+            sin_, cos_ = np.sin(radians[2]), np.cos(radians[2])
+            affine = affine @ np.array([
+                [cos_, -sin_, 0., 0.],
+                [sin_, cos_, 0., 0.],
+                [0., 0., 1., 0.],
+                [0., 0., 0., 1.],
+            ])
+        return affine
+
+    raise ValueError('create_rotate got spatial_dims={}, radians={}.'.format(spatial_dims, radians))
+
+
+def create_shear(spatial_dims, coefs):
+    """
+    create a shearing matrix
+    Args:
+        spatial_dims (int): spatial rank
+        coefs (floats): shearing factors, defaults to 0.
+    """
+    coefs = list(ensure_tuple(coefs))
+    if spatial_dims == 2:
+        while len(coefs) < 2:
+            coefs.append(0.0)
+        return np.array([
+            [1, coefs[0], 0.],
+            [coefs[1], 1., 0.],
+            [0., 0., 1.],
+        ])
+    if spatial_dims == 3:
+        while len(coefs) < 6:
+            coefs.append(0.0)
+        return np.array([
+            [1., coefs[0], coefs[1], 0.],
+            [coefs[2], 1., coefs[3], 0.],
+            [coefs[4], coefs[5], 1., 0.],
+            [0., 0., 0., 1.],
+        ])
+    raise NotImplementedError
+
+
+def create_scale(spatial_dims, s):
```

Comment:
Why not use unified argument name "coefs" for all factor matrix?
I think "s" and "t" are too simple and unclear.
Thanks.